### DISCUSSION TITLE
Download fails on 99%

### DISCUSSION THREAD

**Discussion by BGFR** [aclap-dev/video-downloadhelper#923] [Help and Q&A]
### Your question

HI, i tried several time, but download fails on 99% for videos longer than 30 mins on firefox. And extension shows the alert that video cannot be downloaded with a report button.
Is that intentional? It looks like a limitation?

### Addon & CoApp version and details

version: 9.0.2.12
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:128.0) Gecko/20100101 Firefox/128.0
{
    "used_sidebar_button": true,
    "used_popup_button": true,
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "C:\\Users\\MyPC\\dwhelper"
        }
    },
    "never_show_no_incognito_msg_again": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "yt_warning": false,
    "use_sidebar": true,
    "open_count_store": 6,
    "successfull_dl": 77,
    "never_show_successfull_dl_message": true
}

**Comment by GrampaWildWilly** [comment]
If you want somebody to look into this, you need to supply URLs of at least one, perhaps a few, web pages on which videos are hosted that have been a problem for you.  Do not supply simply the name of a web site.  Provide URLs to specific videos.  These must be pages that do not require a login, even a free one.  The idea is to see whether somebody else has the same problem you do.  So provide enough information for that to happen & let's see where we get with this.

**Comment by BGFR** [comment]
Well the pages require login. I tried on chrome and firefox and the result is the same. It even downloads 100% and after that a window pops up above. (below the image) I reported via that Report button. Could it be that this is a limitation of the free extension? 
![fails](https://github.com/aclap-dev/video-downloadhelper/assets/33952672/d922c020-aef6-49a8-ac7f-09ebd470764e)

**Reply by BGFR** [reply]
I just got the license and result is still the same. Download failed shows up right after 100% reached. The video files shows up in the dwhelper folder but when started shows black screen and 
![fails1](https://github.com/aclap-dev/video-downloadhelper/assets/33952672/07b58be9-59ca-4e00-ba8c-27f1c18989ac)
Also i switched to the legacy view and instead the Download failed window, now it showed the following error:

```
Native port disconnected
Error: Native port disconnected
    at Ca.cleanup (chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:1:43599)
    at chrome-extension://lmjnegcaeklhafolokijcfjliaokphfk/background/main.js:1:42756
```
    
I hope that this will help s.o. to look in to it.

**Reply by Mattmjs** [reply]
Since you have Firefox, do you see this icon in the image. If you do then that would mean it's protected by DRM.

![DRM icon](https://github.com/aclap-dev/video-downloadhelper/assets/160297904/c0abd0b9-5357-47b3-846b-85ea1ed744f5)

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I'm just another user.  Same for Matt.

DRM is worth eliminating as an issue.  But usually with DRM, either VDH will go ahead & give you what looks like a good download but the resulting file plays scrambled junk.  It plays, but it's junk.  Or VDH simply won't recognize anything on the page.  We've got something a bit in between.  It's too bad it's content behind a login.  Most curious that your video player thinks you're trying to do something commercial when it's just a home-user app.

Your idea of buying a license was probably based on faulty assumptions.  The only difference between an unlicensed & a licensed version of VDH is the watermark.  If you don't have a license, certain content will have a watermark.  (And if you're using Chrome, which you are not, you can't do HLS downloads any more often than 2 hours apart.)  The writing of the watermark is usually a painfully lengthy operation even on short videos.  Buying a license does not unlock new features.  It also does not make it work on sites where the unlicensed version fails.  Had you asked me first, I would have told you that buying a license would not have made any difference.  Although I welcome your support of the software, since you just got your license, you can still get a refund for it.  You have 1 month after purchase to ask for a refund.

You might investigate whether your antivirus or firewall or both is/are interfering with the download.  There's been a fair bit of traffic on here on that subject.  Reading some of that might be helpful.

**Reply by BGFR** [reply]
HI, well if DRM i should not be able to download it at all. And also you are right, when i played it with WinMedPlayer, the video is scrambled. May be i can download with another setup? I saw that the classic UI has a lot of options?

**Reply by Mattmjs** [reply]
Whether you can download it or not, if it's DRM you'll have a scrambled file. So if you head back to the video in Firefox &
see that icon I showed you'll know that means it's a protected video.

**Reply by GrampaWildWilly** [reply]
True, it might be better if VDH actually detected DRM.  But it doesn't.  Ffmpeg invoked "manually," meaning from a .bat script or something similar, does the same thing.  It doesn't actually detect DRM.  It just complains about an inability to open key files but downloads what it can anyway.  In general, if the content is protected by DRM, you are not going to download it.  DRM is a way for content creators to prevent people from downloading copyrighted material.  When something is protected by DRM, the client browser is able to decode & play the content, but it can't download it.  Seems self-contradictory but that's the way it works.

You should answer Matt's question.  Do you see the DRM icon on the pages where you can't get useful output?
